### PR TITLE
UR-165 - Reimport initial commit

### PR DESCRIPTION
**Pull request by Batname**

### CODE REVIEW

File: Source/RiveEditor/Private/Factories/RiveFileFactory.cpp [hunk 1]
```diff
@@ -68,3 +77,58 @@ UObject* URiveFileFactory::FactoryCreateFile(UClass* InClass, UObject* InParent,
...
+            RiveFile->RiveFilePath = IFileManager::Get().ConvertToAbsolutePathForExternalAppForRead(*NewReimportPaths[0]);
+        }
+    }
+}
+
+EReimportResult::Type URiveFileFactory::Reimport(UObject* Obj)
+{
+    URiveFile* RiveFile = Cast<URiveFile>(Obj);
+
+    if (!IsValid(Obj)) // || !GDTF->GetGDTFAssetImportData())
```
Comment (Batname) [inline Source/RiveEditor/Private/Factories/RiveFileFactory.cpp:109]: remove this comment

File: Source/RiveEditor/RiveEditor.Build.cs [hunk 2]
```diff
@@ -31,7 +31,8 @@ public RiveEditor(ReadOnlyTargetRules Target) : base(Target)
                 "UnrealEd",
                 "ToolMenus",
                 "UMG",
-                "UMGEditor",
+                "UMGEditor", 
+                "TextureEditor",
```
Comment (Batname) [inline Source/RiveEditor/RiveEditor.Build.cs:35]: do we need this?

File: Source/RiveEditor/Private/RiveFile_AssetTypeActions.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+{
+	return true;
+}
+
+void FRiveFile_AssetTypeActions::OpenAssetEditor(const TArray<UObject*>& InObjects, TSharedPtr<IToolkitHost> EditWithinLevelEditor)
+{
+	if (ensure(InObjects.Num() == 1) && InObjects[0]->IsA<URiveFile>())
+	{
+		ITextureEditorModule* TextureEditorModule = &FModuleManager::LoadModuleChecked<ITextureEditorModule>("TextureEditor");
+		TextureEditorModule->CreateTextureEditor(EToolkitMode::Standalone, EditWithinLevelEditor, Cast<URiveFile>(InObjects[0]));
```
Comment (Batname) [inline Source/RiveEditor/Private/RiveFile_AssetTypeActions.cpp:47]: We have a custom editor where we set rive file artboard, resolution, etc. How that would work

Comment (Batname) [reply Source/RiveEditor/Private/RiveFile_AssetTypeActions.cpp:47]:
```
EAssetCommandResult URiveFile_AssetDefinitionDefault::OpenAssets(const FAssetOpenArgs& OpenArgs) const
{
	for (URiveFile* RiveFile : OpenArgs.LoadObjects<URiveFile>())
	{
		const TSharedRef<FRiveAssetToolkit> EditorToolkit = MakeShared<FRiveAssetToolkit>();
		EditorToolkit->Initialize(RiveFile, OpenArgs.GetToolkitMode(), OpenArgs.ToolkitHost);
	}

	return EAssetCommandResult::Handled;
}

```

Comment (Voulz) [reply Source/RiveEditor/Private/RiveFile_AssetTypeActions.cpp]: Thanks, was looking for this

File: Source/RiveEditor/RiveEditor.Build.cs (hunk 2 above)
Comment (Voulz) [reply Source/RiveEditor/RiveEditor.Build.cs]: Not anymore, thanks

**Pull request merged**